spec/index.bs

```diff
@@ -1073,19 +1069,25 @@ GPUBuffer includes GPUObjectBase;
     : <dfn>\[[mapping]]</dfn> of type {{ArrayBuffer}} or {{Promise}} or `null`.
     ::
         The mapping for this {{GPUBuffer}}.
+
+    : <dfn>\[[mapping_range]]</dfn> of type `sequence<Number>` or `null`.
+    ::
+        The range of this {{GPUBuffer}} that is mapped.
+
+    : <dfn>\[[mapped_ranges]]</dfn> of type `sequence<ArrayBuffer>` or `null`.
```

Comment:
The spec for `getMappedRange` says:

> Let m be a new ArrayBuffer of size size pointing at the content of this.[[mapping]] at offset offset - this.[[mapping_range]][0].

Agreed that it isn't super clear, I'll call out that the mapped ranges can overlap in a note.

Reply:
so what about the act of writing to the same sub-range from different workers?

Reply:
They are `ArrayBuffer` and not `SharedArrayBuffer`. Also there is a TODO to talk about worker restrictions and I added that `getMappedRange` can only be called on the worker on which `mapAsync` was called.

It's an extremely unfortunate restriction, but I don't see a better way to do things in JS. Open to suggestions though.

Reply:
I'm confident this is ok, because you can already do that in JS with the following code:

```js
let a = new ArrayBuffer(12);
let b = new Uint32Array(a, 0, 2);
let c = new Uint32Array(a, 4, 2);
b // Uint32Array(2) [0, 0]
c // Uint32Array(2) [0, 0]
b[1] = 42;
c[0] // 42
```

Reply:
Like you noted, option 1) is not doable so it has to be option 2).